include/sway/tree/node.h

```diff
@@ -17,6 +17,10 @@ enum sway_node_type {
 	N_CONTAINER,
 };
 
+enum {
+	NUM_NODE_TYPES = N_CONTAINER + 1
+};
```

Comment:
This can just be a `const size_t`

Reply:
This compiles:

```c
#include <stdlib.h>

enum abc {
	A,
	B,
	C,
};

const size_t ABC_NUM = C + 1;
```

Reply:
>because it's used in a type definition

Ah right. Yeah, `#define` would be better.

Reply:
Sometimes, the number of elements in an enum is simply added as the last element of the enum. Is that a better option?

```c
enum abc {
	A,
	B,
	C,
	NUM_ABC
};
```

Reply:
I don't like this, because it gets in the way in `switch` statements. It's also not a valid value of the enum.